cli_tools/common/utils/param/param_utils.go

```diff
@@ -0,0 +1,132 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+//+build !test
+
+package paramutils
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/domain"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/storage"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	"google.golang.org/api/option"
+)
+
+// GetProjectID gets project id from either current project or flag
+func GetProjectID(mgce commondomain.MetadataGCEInterface, projectFlag string) (string, error) {
```

Comment:
I don't quite get. It is the original project flag (pointer).

Reply:
It's just about naming parameters: in the context of param_utils, projectFlag is an override value even if running on GCE. I would personally rename it to projectOverride but it's not a big deal either way.

Reply:
I see, yes the initial value of “projectFlag" is used for overriding. Let's keep it because the var itself can be modified, and finally represent the actual project.